### ISSUE TITLE
SOA record creation may be unnecessary, and causes issues with private hosted zones

### ISSUE THREAD

**Issue by kevcube** [cloudposse/terraform-aws-route53-cluster-zone#53]
## Describe the Bug
If you run this module with `-target 'aws_route53_zone.default[0]'` you will get a hosted zone that has NS and SOA records.
Applying the module without such targeting will also create `resource "aws_route53_record" "soa"` which doesn't include any customization options, therefore it isn't currently valuable for this record to be managed in terraform.

If you implement this module for a private hosted zone like #52 allows, then for some reason `resource "aws_route53_record" "soa"` will fail to create.
Therefore I propose that `resource "aws_route53_record" "soa"` is removed from this module.

Unfortunately this will cause deletion of that resource for people who have provisioned this module.

Seeking input from @cloudposse members on how to proceed.

## Expected Behavior
Route53 hosted zones can be created, whether public or private, and will contain an SOA record after creation

## Steps to Reproduce
1. provision module with targeting so that only a hosted zone is created
2. see SOA record created in AWS console
3. destroy module
4. provision module without targeting to create hosted zone and SOA record
5. see SOA record created in AWS console
6. destroy module
7. provision module with VPC as enabled in #52 
8. SOA record will fail to create

**Comment by korenyoni** [comment]
I don't think that the SOA resource should be removed. Simply, private zones should not have SOAs created, because they do indeed fail to create and are only valid for public zones.

In https://github.com/cloudposse/terraform-aws-route53-cluster-zone/pull/52 I've pushed changes to make that happen, however I'm running to other issues relating to the `count` meta attribute on the SOA resource.

**Comment by kevcube** [comment]
> I don't think that the SOA resource should be removed. Simply, private zones should not have SOAs created, because they do indeed fail to create and are only valid for public zones.
I'm curious about the reason behind them ever being included, because AWS seems to automatically create valid NS and SOA resources on both public and private zones. This module then overwrites them (which is not problematic), but with no available customization.

Maybe route53 auto-creating those records was not always the case?

**Comment by korenyoni** [comment]
I'm not entirely sure for the historical reason for this (in terms of this module). @osterman @aknysh @goruha do you know?

**Comment by Nuru** [comment]
@kevcube We create the SOA record so we can change the TTL, which serves as the limit on caching negative responses. Likewise we change the TTL on NS records. We definitely do not want them removed. The proper fix is to suppress the SOA record when creating a private zone.
